test/suites/nodeclaim/garbage_collection_test.go

```diff
@@ -93,6 +95,13 @@ var _ = Describe("NodeClaimGarbageCollection", func() {
 		instanceInput.UserData = lo.ToPtr(base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf(string(rawContent), env.ClusterName,
 			env.ClusterEndpoint, env.ExpectCABundle(), nodePool.Name))))
 
+		instanceProfileName := fmt.Sprintf("KarpenterNodeInstanceProfile-%s", env.ClusterName)
+		ExpectInstanceProfileCreated(instanceProfileName)
+		// Sleep 10 seconds to allow the instance profile to handle eventual consistency delays.
+		time.Sleep(10 * time.Second)
+		DeferCleanup(func() {
+			defer ExpectInstanceProfileDeleted(instanceProfileName)
+		})
```

Comment:
These deferred cleanup methods seem a bit scary and leak prone. When does this block execute?

Reply:
> When DeferCleanup is called in BeforeEach, JustBeforeEach, It, AfterEach, or JustAfterEach the registered callback will be invoked when the spec completes (i.e. it will behave like an AfterEach node)